avm/res/recovery-services/vault/main.bicep

```diff
@@ -411,6 +437,44 @@ output privateEndpoints array = [
 //   Definitions   //
 // =============== //
 
+type customerManagedKeyType = {
```

Comment:
Not aligned with the AVM Interface for CMK: https://azure.github.io/Azure-Verified-Modules/specs/bcp/res/interfaces/#customer-managed-keys

We have a few options here. One would be that you try and merry to expectations of the RSV with the AVM interface to use the same. Next would be that I try to suggest a code change from the top of my head as part of this PR which would need to be tested and finished. Another would be that we merge this PR into a temporal branch in upstream, I apply the updates, open a new PR and have you check the changes. Last but not least, you could give me permissions in your fork to create branches and I could do the same there.
In any case, as it is right now, it is unfortunately not aligned and definitely should before the module is published. From the looks of it, there should be a fair change that we could use the CMK type of the avm-common-types module 1:1 and just need to adjust its implementation.

Reply:
Much appreciated for the help, i will add you to my fork.

Reply:
Great, thanks @alexanderojala. I'll get to it asap